core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -524,9 +524,15 @@ class GroupMetadataManager(brokerId: Int,
           val memRecords = fetchDataInfo.records match {
             case records: MemoryRecords => records
             case fileRecords: FileRecords =>
-              buffer.clear()
-              fileRecords.readInto(buffer, 0)
-              MemoryRecords.readableRecords(buffer)
+              val recordsBuffer = if (fileRecords.sizeInBytes > buffer.capacity) {
+                debug(s"Loaded offsets and group metadata from $topicPartition with buffer larger than configured offsets.load.buffer.size")
+                ByteBuffer.allocate(fileRecords.sizeInBytes)
```

Comment:
Is it ok to allocate a new buffer larger than the configured size for this case? What if we instead logged an error message and threw an exception?

Reply:
Just logging an error and throwing an exception seems fine to me as well. Thought it was fine to temporarily allocate more memory if needed to read a message (even if that makes the `offsets.load.buffer.size` parameter less true, and some systems might run Kafka with tight memory limits) as the above `log.read` has `minOneMessage = true`.
No strong opinion on any side, as long as loading the partition is not silently stuck 🙂

Reply:
Thanks for the explanation. I'm not in a position to have an opinion either!

@ijuma, can someone else review this?